src/dune/super_context.ml

```diff
@@ -602,27 +615,29 @@ module Deps = struct
       Expander.with_record_no_ddeps expander forms ~dep_kind:Optional
         ~map_exe:Fn.id ~c_flags
     in
-    let deps = List.map l ~f:(f t expander) |> Build.all >>^ List.concat in
-    Build.fanout3 deps
-      ( Build.record_lib_deps (Expander.Resolved_forms.lib_deps forms)
-      >>>
-      let ddeps = String.Map.to_list (Expander.Resolved_forms.ddeps forms) in
-      Build.all (List.map ddeps ~f:snd) )
-      (Build.path_set (Expander.Resolved_forms.sdeps forms))
-    >>^ fun (deps, _, _) -> deps
+    let+ deps =
+      Build.map ~f:List.concat (List.map l ~f:(f t expander) |> Build.all)
+    and+ () = Build.record_lib_deps (Expander.Resolved_forms.lib_deps forms)
+    and+ (_ : Value.t list list) =
+      (* TODO: Why do we ignore the resulting [ddeps] values? *)
```

Comment:
I can't remember why this is like this. When we figure out why, we should add a comment here